gfx_scene.h

```diff
@@ -1690,177 +1730,164 @@ class GfxSceneInternal
-                    it = std::find_if(gltf_primitive.attributes, gltf_primitive_attributes_end,
-                        [&](const cgltf_attribute& x) { return x.type == cgltf_attribute_type_texcoord; });  // locate uv stream
-                    cgltf_accessor const *uv_buffer = it != gltf_primitive_attributes_end ? it->data : nullptr;
-                    it = std::find_if(gltf_primitive.attributes, gltf_primitive_attributes_end,
-                        [&](const cgltf_attribute& x) { return x.type == cgltf_attribute_type_joints; });  // joints stream
-                    cgltf_accessor const *joints_buffer = it != gltf_primitive_attributes_end ? it->data : nullptr;
-                    it = std::find_if(gltf_primitive.attributes, gltf_primitive_attributes_end,
-                        [&](const cgltf_attribute& x) { return x.type == cgltf_attribute_type_weights; });  // weights stream
-                    cgltf_accessor const *weights_buffer = it != gltf_primitive_attributes_end ? it->data : nullptr;
-                    std::vector<cgltf_float> sparse_position_buffer;
-                    std::vector<cgltf_float> sparse_normal_buffer;
-                    std::vector<cgltf_float> sparse_uv_buffer;
-                    std::vector<cgltf_float> sparse_joints_buffer;
-                    std::vector<cgltf_float> sparse_weights_buffer;
-                    std::vector<cgltf_float> sparse_index_buffer;
-                    auto unpack_sparse = [](cgltf_accessor const *buffer,
-                                            std::vector<cgltf_float> &sparse_buffer) -> bool {
-                        if(buffer != nullptr && buffer->is_sparse)
-                        {
-                            cgltf_size buffer_size = cgltf_num_components(buffer->type) * buffer->count;
-                            sparse_buffer.resize(buffer_size);
-                            if(cgltf_accessor_unpack_floats(buffer, &sparse_buffer[0], buffer_size) < buffer_size)
-                            {
-                                GFX_PRINT_ERROR(kGfxResult_InternalError, "Failed to unpack sparse accessor");
-                                return false;
-                            }
-                        }
-                        return true;
-                    };
-                    bool failed_unpack = !unpack_sparse(position_buffer, sparse_position_buffer);
-                    failed_unpack      = failed_unpack || !unpack_sparse(normal_buffer, sparse_normal_buffer);
-                    failed_unpack      = failed_unpack || !unpack_sparse(uv_buffer, sparse_uv_buffer);
-                    failed_unpack      = failed_unpack || !unpack_sparse(joints_buffer, sparse_joints_buffer);
-                    failed_unpack      = failed_unpack || !unpack_sparse(weights_buffer, sparse_weights_buffer);
-                    failed_unpack      = failed_unpack || !unpack_sparse(index_buffer, sparse_index_buffer);
```

Comment:
Any reason for removing these `unpack_sparse()` calls? They don't seem to appear anywhere in the new code.

Reply:
`UnpackAccessor` just below does the same thing. Although it now always uses the unpack floats function for sparse and non-sparse accessors. It wasnt done that way previously and I have no idea why, I think we had a sparse/non-sparse code path for performance reasons but I cant remeber. It would be interesting to know what performance difference it makes having only a single code path

Reply:
Well I added `UnpackAccessor` in order to fix this comment. And I decided to always do unpacking since it's just much simpler in terms of implementation and readability. I guess it might have negative impact on performance, but it's surely not too large and is limited to model loading time. Unfortunately I just don't have large models with sparse accessors to test it.